src/ctap/mod.rs

```diff
@@ -274,7 +274,7 @@ fn send_keepalive_up_needed(
     let keepalive_msg = CtapHid::keepalive(cid, KeepaliveStatus::UpNeeded);
     for mut pkt in keepalive_msg {
         let ctap_hid_connection = transport.hid_connection(env);
-        match ctap_hid_connection.send_or_recv_with_timeout(&mut pkt, timeout) {
+        match ctap_hid_connection.send_and_maybe_recv(&mut pkt, timeout) {
```

Comment:
Maybe we have to add more HID tests? The most common situation with unexpected OUT packets (and therefore OUT and IN at the same time) is probably a CANCEL during user presence checks. Should be something like a MakeCredential request, and while user presence is checked, spam it with packets, and see if those are treated correctly.

Reply:
Agreed. Added these tests.